### PR TITLE
Add role for peer-probe and peer-detach

### PR DESCRIPTION
**Pull request by siddharthvipul**
- modules/peer.py
+ playbooks/gluster-peer-detach.yml
+ playbooks/gluster-peer-probe.yml
+ modules/gluster_peer.py
+ roles/

### CODE REVIEW

File: examples/peer-detach-force.conf [hunk 1]
```diff
@@ -9,8 +9,8 @@
 # force section has 'no' as its default value.
 
 [hosts]
-10.70.46.13
-10.70.46.15
+192.168.122.8
+192.168.122.151
```
Comment (sac) [inline examples/peer-detach-force.conf]: These changes are not needed. Why did you do these changes?

File: modules/gluster_peer.py [hunk 2]
```diff
@@ -18,6 +18,40 @@
...
+        description: When the action variable is set to probe,
+        the remote machines listed in the hosts sections are added to the Trusted Storage Pool.
+        When the action variable is set to detach, 
+        the remote machines listed in the hosts sections are removed from the Trusted Storage Pool. 
+        required: true
+        choices: [add/probe, remove/detach]
+        
+    master:
+        description: current node so that it can be removed from hosts' list (if available)
+
```
Comment (sac) [inline modules/gluster_peer.py:45]: Make sure that the lines do not cross 80 column mark.

File: roles/peer_detach/README.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+License
+-------
+
+BSD
+
+Author Information
+------------------
+
+An optional section for the role authors to include contact information, or a website (HTML is not allowed).
```
Comment (sac) [inline roles/peer_detach/README.md:38]: Clean up this file, you have left the template as is.

File: roles/peer_detach/meta/main.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    # List tags for your role here, one per line. A tag is a keyword that describes
+    # and categorizes the role. Users find roles by searching for tags. Be sure to
+    # remove the '[]' above, if you add tags to this list.
+    #
+    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
+    #       Maximum 20 tags per role.
+
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
```
Comment (sac) [inline roles/peer_detach/meta/main.yml:57]: Same here.

File: roles/peer_detach/tests/test.yml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - peer_detach
```
Comment (sac) [inline roles/peer_detach/tests/test.yml:5]: Have you tested this?

File: roles/peer_probe/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+License
+-------
+
+BSD
+
+Author Information
+------------------
+
+An optional section for the role authors to include contact information, or a website (HTML is not allowed).
```
Comment (sac) [inline roles/peer_probe/README.md:38]: This needs cleanup too.

File: roles/peer_probe/meta/main.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    # List tags for your role here, one per line. A tag is a keyword that describes
+    # and categorizes the role. Users find roles by searching for tags. Be sure to
+    # remove the '[]' above, if you add tags to this list.
+    #
+    # NOTE: A tag is limited to a single word comprised of alphanumeric characters.
+    #       Maximum 20 tags per role.
+
+dependencies: []
+  # List your role dependencies here, one per line. Be sure to remove the '[]' above,
+  # if you add dependencies to this list.
```
Comment (sac) [inline roles/peer_probe/meta/main.yml:57]: This needs clean up too.

File: roles/peer_probe/tests/test.yml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+---
+- hosts: localhost
+  remote_user: root
+  roles:
+    - peer_probe
```
Comment (sac) [inline roles/peer_probe/tests/test.yml:5]: Have you tested this?

**Review by sac — CHANGES_REQUESTED** [review]

**Comment by siddharthvipul** [comment]
I will change everything required and get back to you

File: examples/peer-detach-force.conf (hunk 1 above)
Comment (siddharthvipul) [reply examples/peer-detach-force.conf]: I don't have vms to test. so I was testing it in my local vms

**Pull request closed without merging**